### ISSUE TITLE
Issues during installation of rsyslog example role on fedora 28

### ISSUE THREAD

**Issue by sradco** [linux-system-roles/logging#12]
I run the rsyslog role on Fedora28.
I run it with the example rsyslog vars.yaml.
```
rsyslog__enabled: true
# install example packages & config files
rsyslog__example: true
rsyslog__capabilities: [ 'network', 'remote-files', 'tls' ]
rsyslog__forward: [ '' ]
rsyslog__group: root
rsyslog__user: root
```
I had a few issues:

1. I was missing the python2-libselinux package
TASK [rsyslog : Generate main rsyslog configuration] *****************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "checksum": "6d5d22e905ff75f31050d24cc0adb3ca18f57430", "msg": "Aborting, target uses selinux but python bindings (libselinux-python) aren't installed!"}

2. TASK [rsyslog : Install/Update required packages] ********************************************************************************************************************************************
[DEPRECATION WARNING]: Invoking "yum" only once while using a loop via squash_actions is deprecated. Instead of using a loop to supply multiple items and specifying `name: {{ item }}`, 
please use `name: [u'{{ rsyslog__base_packages }}', u'{{ rsyslog__viaq_prereq_packages if rsyslog__viaq|bool else [] }}', u'{{ rsyslog__viaq_packages if rsyslog__viaq|bool else [] }}', u'{{
 rsyslog__tls_packages if rsyslog__pki|bool else [] }}', u'{{ rsyslog__packages }}']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled
 by setting deprecation_warnings=False in ansible.cfg.

3. TASK [rsyslog : Moving the contents of /etc/rsyslog.d to the backup dir] *********************************************************************************************************************
 [WARNING]: Consider using the unarchive module rather than running tar.  If you need to use command because unarchive is insufficient you can add warn=False to this command task or set
command_warnings=False in ansible.cfg to get rid of this message.

@richm @nhosoi

**Comment by nhosoi** [comment]
Regarding 1:
I already had libselinux-python on my test host, which is needed by multiple packages [1].  I forced to remove libselinux-python, then I could reproduce the problem.  If I add libselinux-python in rsyslog__base_packages as in [2], we won't run into the failure.  But I have an impression there could be more appropriate package(s) to be there which would install all the necessary packages including 'libselinux-python'...  Do you have any good idea for such candidates?  For instance, libselinux-python has these direct dependencies in my test env... [3]

```
# rpm -q libselinux-python
libselinux-python-2.5-14.1.el7.x86_64
```

[1] 
```
# rpm -e libselinux-python
error: Failed dependencies:
    libselinux-python is needed by (installed) openshift-ansible-3.9.1-1.git.0.f99af1f.el7.centos.noarch
    libselinux-python is needed by (installed) python-slip-0.4.0-4.el7.noarch
    libselinux-python >= 2.1.5-1 is needed by (installed) setroubleshoot-server-3.2.29-3.el7.x86_64
    libselinux-python is needed by (installed) anaconda-core-21.48.22.134-1.el7.centos.x86_64
    libselinux-python is needed by (installed) policycoreutils-python-2.5-29.el7.x86_64

```
[2] 
```
# diff --git a/roles/rsyslog/defaults/main.yaml b/roles/rsyslog/defaults/main.yaml
index c91694e..fcbefed 100644
--- a/roles/rsyslog/defaults/main.yaml
+++ b/roles/rsyslog/defaults/main.yaml
@@ -61,7 +61,7 @@ rsyslog__message_reduction: False
 # .. envvar:: rsyslog__base_packages
 #
 # List of default rpm packages to install.
-rsyslog__base_packages: [ 'rsyslog' ]
+rsyslog__base_packages: [ 'rsyslog', 'libselinux-python' ]
 
 # .. envvar:: rsyslog__logging_packages
 #
```

[3] 
```
# rpm -e libselinux-python
error: Failed dependencies:
    libselinux-python is needed by (installed) openshift-ansible-3.9.1-1.git.0.f99af1f.el7.centos.noarch
    libselinux-python is needed by (installed) python-slip-0.4.0-4.el7.noarch
    libselinux-python is needed by (installed) anaconda-core-21.48.22.134-1.el7.centos.x86_64
    libselinux-python is needed by (installed) policycoreutils-python-2.5-29.el7.x86_64
    libselinux-python >= 2.1.5-1 is needed by (installed) setroubleshoot-server-3.2.30-3.el7.x86_64
```

**Comment by nhosoi** [comment]
@sradco, could you take a look at https://github.com/linux-system-roles/logging/pull/20 ?
Thanks!

**Comment by sradco** [comment]
> Regarding 1:
> I already had libselinux-python on my test host, which is needed by multiple packages [1]. I forced to remove libselinux-python, then I could reproduce the problem. If I add libselinux-python in rsyslog__base_packages as in [2], we won't run into the failure. But I have an impression there could be more appropriate package(s) to be there which would install all the necessary packages including 'libselinux-python'... Do you have any good idea for such candidates? For instance, libselinux-python has these direct dependencies in my test env... [3]
> 
> ```
> # rpm -q libselinux-python
> libselinux-python-2.5-14.1.el7.x86_64
> ```
> [1]
> 
> ```
> # rpm -e libselinux-python
> error: Failed dependencies:
>     libselinux-python is needed by (installed) openshift-ansible-3.9.1-1.git.0.f99af1f.el7.centos.noarch
>     libselinux-python is needed by (installed) python-slip-0.4.0-4.el7.noarch
>     libselinux-python >= 2.1.5-1 is needed by (installed) setroubleshoot-server-3.2.29-3.el7.x86_64
>     libselinux-python is needed by (installed) anaconda-core-21.48.22.134-1.el7.centos.x86_64
>     libselinux-python is needed by (installed) policycoreutils-python-2.5-29.el7.x86_64
> ```
> [2]
> 
> ```
> # diff --git a/roles/rsyslog/defaults/main.yaml b/roles/rsyslog/defaults/main.yaml
> index c91694e..fcbefed 100644
> --- a/roles/rsyslog/defaults/main.yaml
> +++ b/roles/rsyslog/defaults/main.yaml
> @@ -61,7 +61,7 @@ rsyslog__message_reduction: False
>  # .. envvar:: rsyslog__base_packages
>  #
>  # List of default rpm packages to install.
> -rsyslog__base_packages: [ 'rsyslog' ]
> +rsyslog__base_packages: [ 'rsyslog', 'libselinux-python' ]
>  
>  # .. envvar:: rsyslog__logging_packages
>  #
> ```
> [3]
> 
> ```
> # rpm -e libselinux-python
> error: Failed dependencies:
>     libselinux-python is needed by (installed) openshift-ansible-3.9.1-1.git.0.f99af1f.el7.centos.noarch
>     libselinux-python is needed by (installed) python-slip-0.4.0-4.el7.noarch
>     libselinux-python is needed by (installed) anaconda-core-21.48.22.134-1.el7.centos.x86_64
>     libselinux-python is needed by (installed) policycoreutils-python-2.5-29.el7.x86_64
>     libselinux-python >= 2.1.5-1 is needed by (installed) setroubleshoot-server-3.2.30-3.el7.x86_64
> ```

In Fedora the Fedora-28_updates repo includes this package , but we dont install it anywhere in the code ... I only needed to do yum install libselinux-python to get it.

@nhosoi

**Comment by richm** [comment]
I don't understand what the problem is.  We don't have any direct dependencies on libselinux-python.  If anything, it is a dependency of ansible, the selinux module perhaps.  A bug should be opened against Fedora 28 ansible or maybe ansible-selinux if there is a separate package for the selinux functionality in ansible.

**Comment by sradco** [comment]
The problem was that the package was missing when running the role. When I manually installed it is was available in Fedora repo. So we probably need to add it to the packages list to install no?
@richm

**Comment by richm** [comment]
> The problem was that the package was missing when running the role. When I manually installed it is was available in Fedora repo. So we probably need to add it to the packages list to install no?

Ok, but that is a workaround, not a fix.  The fix is to identify the package that depends on it, and update the `Requires` in the rpm spec for that package, so that `libselinux-python` is properly pulled in.

**Comment by richm** [comment]
>  ToDo: We should revert this patch and find out the package which requires libselinux-python, then file a bug against the package. Correct?

Correct, although we may have to leave the `libselinux-python` in the list until we can fix the problem with the package.

**Comment by nhosoi** [comment]
@richm ,
Sadly, it's already done in bz and it was closed with NOTABUG... :(
https://bugzilla.redhat.com/show_bug.cgi?id=1505083#c3
(and we could find lots of "issues" in the ansible github...)
For us, it'd be safer to have it in rsyslog_base_packages...?

**Comment by richm** [comment]
I don't understand the reasoning in the bz, but ok - let's just add the package in rsyslog_base_packages

**Comment by nhosoi** [comment]
@sradco, can we consider this "issue" was solved in pr/#20?
